src/test/scala/ai/privado/tagger/sink/SinkArgumentUtilityTest.scala

```diff
@@ -17,136 +18,238 @@ import io.shiftleft.codepropertygraph.generated.Cpg
+        // Retrieve the corresponding value from the argumentsList
+        val argumentsListValue = argumentsList.getOrElse(key, "")
+        assert(argumentsListValue == value, s"Expected value '$value' for key '$key', but got '$argumentsListValue'")
+      }
+    }
+
+    "Check nested field arguments" in {
+      val cpg = code("""
+          |globalThis.dataLayer.push({
+          |    nestedKey: {
+          |        nested1: "N1",
+          |        nestedDeep: {
+          |            nestedDeep1: "N1.11"
+          |        }
+          |    }
+          |});
+          |""".stripMargin)
+      val expectedArgumentsMap = Map("nestedKey.nested1" -> "\"N1\"", "nestedKey.nestedDeep.nestedDeep1" -> "\"N1.11\"")
+
+      val gaSinkNode = cpg.call.name("push").head
+      gaSinkNode.tag.nameExact(Constants.id).value.headOption shouldBe Some(Constants.googleTagManagerPixelRuleId)
+      val encodedArgumentsList = gaSinkNode.tag.nameExact(Constants.arguments).value.headOption.getOrElse("{}")
+      val argumentsList        = SinkArgumentUtility.deserializedArgumentString(encodedArgumentsList)
+
+      // Iterate over each key-value pair in the argumentsMap
+      expectedArgumentsMap.foreach { case (key, value) =>
+        // Retrieve the corresponding value from the argumentsList
+        val argumentsListValue = argumentsList.getOrElse(key, "")
+        assert(argumentsListValue == value, s"Expected value '$value' for key '$key', but got '$argumentsListValue'")
+      }
     }
+
+    "Check pick call node arguments" in {
+      val cpg = code("""
+          |import { pick } from 'lodash';
+          |globalThis.dataLayer.push({
+          |    latitude: obj.latitude,
+          |    ...pick(product, ['id', 'name', 'ext_id', 'ic_product_id'])
+          |});
+          |""".stripMargin)
+      val expectedArgumentsMap = Map(
+        "latitude"           -> "obj.latitude",
+        ".\"id\""            -> "product.\"id\"",
+        ".\"ext_id\""        -> "product.\"ext_id\"",
+        "product"            -> "ANY",
+        ".pick.lodash"       -> "",
+        ".\"ic_product_id\"" -> "product.\"ic_product_id\"",
+        ".\"name\""          -> "product.\"name\""
+      )
+
+      val gaSinkNode = cpg.call.name("push").head
+      gaSinkNode.tag.nameExact(Constants.id).value.headOption shouldBe Some(Constants.googleTagManagerPixelRuleId)
+      val encodedArgumentsList = gaSinkNode.tag.nameExact(Constants.arguments).value.headOption.getOrElse("{}")
+      val argumentsList        = SinkArgumentUtility.deserializedArgumentString(encodedArgumentsList)
+      // Iterate over each key-value pair in the argumentsMap
+      expectedArgumentsMap.foreach { case (key, value) =>
+        // Retrieve the corresponding value from the argumentsList
+        val argumentsListValue = argumentsList.getOrElse(key, "")
+        assert(argumentsListValue == value, s"Expected value '$value' for key '$key', but got '$argumentsListValue'")
```

Comment:
I see these 4-5 lines being repeated across, create a method and use that instead